app/overnight/purge-soft-deleted-records.js

```diff
@@ -0,0 +1,152 @@
+const { paranoidRetentionPeriod } = require('../config/index')
+const { overnightJobUser } = require('../constants/import')
+
+const sequelize = require('../config/db')
+const { Op } = require('sequelize')
+const { purgeDogByIndexNumber } = require('../repos/dogs')
+const { purgePersonByReferenceNumber } = require('../repos/people')
+
+/**
+ * Class representing a count of deleted records.
+ * @constructor
+ */
+function DeletedRecordCount () {
+  /**
+   * @type {number}
+   */
+  this.dogs = 0
+  /**
+   * @type {number}
+   */
+  this.owners = 0
+}
+
+/**
+ * Get the total number of deleted records.
+ * @name DeletedRecordCount#total
+ * @type {number}
+ * @readonly
+ */
+Object.defineProperty(DeletedRecordCount.prototype, 'total', {
+  get () {
+    return this.dogs + this.owners
+  }
+})
+
+/**
+ * @constructor
+ */
+function DeletedRecords () {
+  /**
+   * @type {string[]}
+   */
+  this.dogs = []
+  /**
+   * @type {string[]}
+   */
+  this.owners = []
+}
+
+/**
+ * @typedef {{
+ *  count: {
+ *    success: DeletedRecordCount,
+ *    failed: DeletedRecordCount
+ *  },
+ *  deleted: {
+ *    success: DeletedRecords,
+ *    failed: DeletedRecords
+ *  }
+ * }} SoftDeletedRecordPayload
+ */
+
+/**
+ * @param {Date} date
+ * @return {Promise<SoftDeletedRecordPayload>}
+ */
+const purgeSoftDeletedRecords = async (date = new Date()) => {
+  try {
+    const deleteAtCutOff = new Date(date)
+    deleteAtCutOff.setUTCDate(deleteAtCutOff.getUTCDate() - paranoidRetentionPeriod)
+
+    const ownerResults = await sequelize.models.person.findAll({
+      where: {
+        deleted_at: {
+          [Op.lte]: deleteAtCutOff
```

Comment:
Yes, you correctly don't need a null check here

Reply:
I'll add this, can't see it doing any harm to add an extra index